File: Ejercicios Propuestos 3/Ejercicio 9/Ejercicio 9.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Solicitar el tamaño de las matrices
n = int(input("Ingrese el tamaño (n) de las matrices cuadradas: "))

# Crear las matrices A, B y la matriz resultado
matrizA = [[0 for c in range(n)] for f in range(n)]
matrizB = [[0 for c in range(n)] for f in range(n)]
resultado = [[0 for c in range(n)] for f in range(n)]

# Leer los elementos de la matriz A
print("\nELEMENTOS DE LA MATRIZ (A)")
for f in range(n):
    for c in range(n):
        matrizA[f][c] = int(input(f"Elemento A[{f}][{c}]: "))

# Leer los elementos de la matriz B
print("\nELEMENTOS DE LA MATRIZ (B)")
for f in range(n):
    for c in range(n):
        matrizB[f][c] = int(input(f"Elemento B[{f}][{c}]: "))

# Multiplicación de matrices
for f in range(n):
    for c in range(n):
        resultado[f][c] = 0

        # Multiplicar la fila de A por la columna de B
        for k in range(n):
            resultado[f][c] += matrizA[f][k] * matrizB[k][c]

# Mostrar la matriz A
print("\nMATRIZ (A):")
for f in range(n):
    for c in range(n):
        print(matrizA[f][c], end="  ")
    print()

# Mostrar la matriz B
print("\nMATRIZ (B):")
for f in range(n):
    for c in range(n):
        print(matrizB[f][c], end="  ")
    print()

# Mostrar la matriz resultado
print("\nMATRIZ (A × B):")
for f in range(n):
    for c in range(n):
        print(resultado[f][c], end="  ")
    print()

File: Ejercicios Propuestos 3/Ejercicio 9/ejercicio9.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cout << "Ingrese el tamano de las matrices cuadradas: ";
    cin >> n;

    int A[20][20], B[20][20], C[20][20];

    cout << "Ingrese la matriz A:" << endl;
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            cin >> A[i][j];

    cout << "Ingrese la matriz B:" << endl;
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            cin >> B[i][j];

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            C[i][j] = 0;
            for (int k = 0; k < n; k++) {
                C[i][j] = C[i][j] + A[i][k] * B[k][j];
            }
        }
    }

    cout << "Matriz resultado C:" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << C[i][j] << " ";
        }
        cout << endl;
    }
    return 0;
}
